### PR TITLE
Load XML docs when using #r directives in scripting

### LINKED ISSUE #1026: Editing .csx: XML Docs of own libraries are not displayed in Intellisense popups
_From @KarloX2 on November 13, 2017 5:44_

## Environment data
`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-003131)

Product Information:
 Version:            1.0.0-preview2-003131
 Commit SHA-1 hash:  635cf40e58

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.15063
 OS Platform: Windows
 RID:         win10-x64

VS Code version: 1.18.0
C# Extension version: 1.13.0

## Steps to reproduce

Create a **.csx** files in VS Code and put the line

    #r "C:\Some\Folder\MyLibrary.dll"

in the first line. Make sure that, in the same folder, a .xml documentation file (crated by Visual Studio along with the .dll) exists. Assume there's a class MyClass defined in the dll and documented in the .xml. Try to declare a variable of that class type and observe that the documentation is _not_ being displayed in the IntelliSense popup.

    MyClass myVariable; //<<<--- IntelliSense will show signatures but not the docs.

## Expected  behavior

Member documentation shall be included in IntelliSense popups.

## Actual behavior

No member documentation is displayed.

_Copied from original issue: OmniSharp/omnisharp-vscode#1858_

### PR DESCRIPTION
**Pull request by filipw**
Fixes #1026 

At the moment, the `RuntimeMetadataReferenceResolver` on which we rely when processing `#r` directives in scripts, completely ignores XML docs https://github.com/dotnet/roslyn/blob/version-2.4.0/src/Scripting/Core/Hosting/Resolvers/RuntimeMetadataReferenceResolver.cs#L69

There is no simple hook in which we can inject it (unless we rewrite/copy entire resolver and all its dependencies), so the fix is applied via 😔 reflection 😔. I'll talk to Tomas to see if we can do something better on the Scripting API side, but for now this should at least fix the problem.

Also attached a test that verifies the functionality.

### CODE REVIEW

File: src/OmniSharp.Script/ScriptHelper.cs [hunk 1]
```diff
@@ -7,11 +7,19 @@
 using Microsoft.CodeAnalysis.Scripting;
 using Microsoft.CodeAnalysis.Scripting.Hosting;
 using Microsoft.Extensions.Configuration;
+using System.IO;
```
Comment (DustinCampbell) [inline src/OmniSharp.Script/ScriptHelper.cs]: nit: sort System using directives on top.

File: tests/OmniSharp.Roslyn.CSharp.Tests/TypeLookupFacts.cs [hunk 2]
```diff
@@ -5,6 +5,7 @@
 using TestUtility;
 using Xunit;
 using Xunit.Abstractions;
+using System.IO;
```
Comment (DustinCampbell) [inline tests/OmniSharp.Roslyn.CSharp.Tests/TypeLookupFacts.cs]: Sort usings with System namespaces an top

File: src/OmniSharp.Script/ScriptProjectSystem.cs [hunk 3]
```diff
@@ -215,6 +215,7 @@ private void AddMetadataReference(ISet<MetadataReference> referenceCollection, s
             _logger.LogDebug($"Added reference to '{fileReference}'");
         }
 
+
```
Comment (DustinCampbell) [inline src/OmniSharp.Script/ScriptProjectSystem.cs]: nit: Extra blank line

File: src/OmniSharp.Script/ScriptHelper.cs [hunk 4]
```diff
@@ -108,5 +115,27 @@ public ProjectInfo CreateProject(string csxFileName, IEnumerable<MetadataReferen
...
+                {
+                    var documentationFile = Path.ChangeExtension(path, ".xml");
+                    var documentationProvider = File.Exists(documentationFile)
+                        ? XmlDocumentationProvider.CreateFromFile(documentationFile)
+                        : null;
+
+                    return MetadataReference.CreateFromFile(path, properties, documentationProvider);
+                }));
+            }
+
```
Comment (DustinCampbell) [inline src/OmniSharp.Script/ScriptHelper.cs]: nit: extra blank line.

File: src/OmniSharp.Script/ScriptHelper.cs [hunk 5]
```diff
@@ -88,8 +94,9 @@ private CachingScriptMetadataResolver CreateMetadataReferenceResolver()
                 }
             }
             
-            return enableScriptNuGetReferences ? new CachingScriptMetadataResolver(new NuGetMetadataReferenceResolver(ScriptMetadataResolver.Default)) 
-                : new CachingScriptMetadataResolver(ScriptMetadataResolver.Default);
+            return enableScriptNuGetReferences ?
+                new CachingScriptMetadataResolver(new NuGetMetadataReferenceResolver(_resolver)) :
+                new CachingScriptMetadataResolver(_resolver);
```
Comment (DustinCampbell) [inline src/OmniSharp.Script/ScriptHelper.cs]: nit: for this conditional expression, you have the `?` and `:` at the end of the lines. However, below, you have them at the start of the lines. Maybe make these consistent? (I personally prefer the bottom one.)

File: src/OmniSharp.Roslyn/Services/MetadataFileReferenceCache.cs [hunk 6]
```diff
@@ -47,7 +47,7 @@ public MetadataReference GetMetadataReference(string filePath)
             }
 
             var displayText = assemblyMetadata.GetModules().FirstOrDefault()?.Name;
-
+
```
Comment (DustinCampbell) [inline src/OmniSharp.Roslyn/Services/MetadataFileReferenceCache.cs]: unintentional whitespace-only change?

**Comment by filipw** [comment]
thanks! pushed the requested changes. Also added a changelog entry that I forgot before.

**Pull request merged**